src/test/scala/codacy/codenarc/CodeNarcSpec.scala

```diff
@@ -28,5 +28,49 @@ class CodeNarcSpec extends AnyWordSpec with Matchers {
 
       CodeNarc.ruleFileContentFromPatterns(patterns) should be(expectedValue)
     }
+
+    "return a valid CodeNarc rule file with parameter values having regex values" in {
+      val regex1 = ".*(Spec|Test|Tests|TestCase)\\.groovy" // before parsing -> [.*(Spec|Test|Tests|TestCase)\.groovy]
+      val regex2 = "I\\u2019m" // before parsing -> ["I\u2019m"]
+      val patterns: List[CodeNarc.Pattern] = {
+        List(
+          CodeNarc.Pattern(
+            "AssignmentInConditional",
+            Set(PatternParameter("Example", JsNumber(2)), PatternParameter("Param", JsString(regex1)))
+          ),
+          CodeNarc.Pattern(
+            "BigDecimalInstantiation",
+            Set(PatternParameter("Example", JsNumber(2)), PatternParameter("Param", JsString(regex2)))
+          )
+        )
+      }
+
+      val expectedValue =
+        s"""ruleset {
+           |AssignmentInConditional {
+           |Example = 2
+           |Param = '.*(Spec|Test|Tests|TestCase)\\\\.groovy'
+           |}
+           |BigDecimalInstantiation {
+           |Example = 2
+           |Param = 'I\\\\u2019m'
+           |}
+           |}""".stripMargin
```

Comment:
This is what we should pass to the tool. i had to put 4 backslashes to represent 2 (\\).